### PR TITLE
refactor: Refactor code and component structure

### PR DESCRIPTION
**Pull request by chenxi24**

### CODE REVIEW

File: packages/devui-vue/docs/en-US/components/editable-select/index.md [hunk 1]
```diff
@@ -140,42 +306,41 @@ export default defineComponent({
...
     <template #item="slotProps">
-      <div>第{{ slotProps.value }}项: {{ slotProps.label }}</div>
+      <div>第{{ slotProps.index }}项: {{ slotProps.option.label }}</div>
     </template>
-    <template #noResultItem>
-      <div>
-        {{ `没有匹配项` }}
-      </div>
+    <template>
+      <div>没有匹配项</div>
```
Comment (GaoNeng-wWw) [inline packages/devui-vue/docs/en-US/components/editable-select/index.md:319]:
```suggestion
      <div>No matches</div>
```

File: packages/devui-vue/docs/en-US/components/editable-select/index.md [hunk 2]
```diff
@@ -140,42 +306,41 @@ export default defineComponent({
...
+You can customize how individual options are rendered and display templates when no matching data is available.
 :::demo
 
 ```vue
 <template>
-  <d-editable-select v-model="value" :width="450" filter-option :options="options">
+  <d-editable-select v-model="value" :width="450" :options="options">
     <template #item="slotProps">
-      <div>第{{ slotProps.value }}项: {{ slotProps.label }}</div>
+      <div>第{{ slotProps.index }}项: {{ slotProps.option.label }}</div>
```
Comment (GaoNeng-wWw) [inline packages/devui-vue/docs/en-US/components/editable-select/index.md:316]:
```suggestion
      <div>Item {{ slotProps.index }}: {{ slotProps.option.label }}</div>
```

**Review by GaoNeng-wWw — CHANGES_REQUESTED** [review]
我做了一些拼写与翻译上的检查👍

File: packages/devui-vue/docs/en-US/components/editable-select/index.md (hunk 2 above)
Comment (chenxi24) [reply packages/devui-vue/docs/en-US/components/editable-select/index.md:316]: 好的，马上改。

**Review by GaoNeng-wWw — APPROVED** [review]

**Review by GaoNeng-wWw — APPROVED** [review]

**Review by GaoNeng-wWw — APPROVED** [review]

**Review by GaoNeng-wWw — APPROVED** [review]

**Review by GaoNeng-wWw — APPROVED** [review]

**Review by GaoNeng-wWw — APPROVED** [review]

**Comment by GaoNeng-wWw** [comment]
@chenxi24 

`The base branch requires all commits to be signed.`记得[为commit签名](https://cloud.tencent.com/developer/article/2111579)哦~，否则无法合并

**Comment by kagol** [comment]
> @chenxi24
> 
> `The base branch requires all commits to be signed.`记得[为commit签名](https://cloud.tencent.com/developer/article/2111579)哦~，否则无法合并

可以参考下：https://insights.thoughtworks.cn/how-to-sign-git-commit/

**Pull request closed without merging**